src/Features/Core/Portable/GoToDefinition/IGoToDefinitionSymbolService.cs

```diff
@@ -12,5 +12,11 @@ namespace Microsoft.CodeAnalysis.GoToDefinition
     internal interface IGoToDefinitionSymbolService : ILanguageService
     {
         Task<(ISymbol?, TextSpan)> GetSymbolAndBoundSpanAsync(Document document, int position, bool includeType, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// If the position is on a continue/break keyword, returns a token of the corresponding loop statement.
+        /// Otherwise, returns null.
+        /// </summary>
+        Task<SyntaxToken?> GetTargetIfControlFlowAsync(Document document, int position, CancellationToken cancellationToken);
```

Comment:
i'd more expect just a position to be returned rather than a token.  Or, if a token is returned, we just navigate to the start of it.